src/components/Switch/Switch.tsx

```diff
@@ -1,184 +1,231 @@
-                      },
-        };
-    }, [componentStyles, disabled, value]);
+    useEffect(() => {
+        Animated.timing(toggleMarginAnimation, {
+            toValue: value ? 1 : 0,
+            duration: 300,
+            easing: Easing.bezier(0.175, 0.885, 0.32, 1.275),
+            useNativeDriver: false,
+        }).start();
+        Animated.timing(toggleSizeAnimation, {
+            toValue: value ? 1 : 0,
+            duration: 250,
+            easing: Easing.ease,
+            useNativeDriver: false,
+        }).start();
+    }, [value, toggleMarginAnimation, toggleSizeAnimation]);
 
-    return (
-        <NativeSwitch
-            ref={ref}
-            value={value}
-            disabled={disabled}
-            onValueChange={disabled ? undefined : onValueChange}
-            style={switchStyle}
-            {...props}
-            {...rest}
-        />
+    const thumbOverlay = useMemo(
+        () => ({
+            position: 'absolute',
+            left:
+                value && pressed
+                    ? size * -0.21
+                    : value || unCheckedIcon
+                    ? size * -0.28
+                    : !unCheckedIcon && !pressed
+                    ? size * -0.36
+                    : size * -0.16,
+            right: 0,
+            height: size * 1.2,
+            width: size * 1.2,
+            borderRadius: (size * 1.2) / 2,
+            backgroundColor: pressed
+                ? 'rgba(0, 0, 0, 0.15)'
+                : focused
+                ? 'rgba(0, 0, 0, 0.1)'
+                : 'rgba(0, 0, 0, 0.05)',
+            display: hovered || focused || pressed ? 'flex' : 'none',
+        }),
+        [focused, hovered, pressed, size, unCheckedIcon, value],
     );
-};
 
-type CustomProp = { checkedColor?: string; thumbTintColor?: string; onTintColor?: string };
+    const { parentViewStyle, buttonStyle, toggleStyle, toggleWheelStyle, overlayStyle, iconStyle } =
+        useMemo(() => {
+            const { button, toggle, toggleWheel, icon } = componentStyles;
+            return {
+                buttonStyle: [button, size && { borderRadius: size / 2 }],
+                parentViewStyle: {
+                    width: size,
+                    height: size,
+                    borderRadius: size / 2,
+                    flexDirection: 'row',
+                    alignItems: 'center',
+                },
+                toggleStyle: [
+                    toggle,
+                    color && { backgroundColor: color },
+                    {
+                        height: size,
+                        width: size * 1.62,
+                        borderRadius: size / 1.62,
+                        borderWidth: Math.floor(size * 0.095),
+                    },
+                    value && { borderWidth: 0 },
+                ],
+                toggleWheelStyle: [
+                    toggleWheel,
+                    {
+                        marginLeft: moveToggle,
```

Comment:
Why do this with margin?
Can't we use position absolute here?

Reply:
even if we do position, we still have to update left or right style value prop @thetaungg